tests/regression/46-apron2/19-tid-toy-10-exit.c

```diff
@@ -0,0 +1,45 @@
+// SKIP PARAM: --set ana.activated[+] apron --set ana.path_sens[+] threadflag --set ana.activated[+] threadJoins --sets ana.apron.privatization mutex-meet-tid
+// Copy of 80-tid-toy10 making use of pthread_exit
+#include <pthread.h>
+#include <assert.h>
+
+int g = 10;
+int h = 10;
+pthread_mutex_t A = PTHREAD_MUTEX_INITIALIZER;
+
+void *t_benign(void *arg) {
+  int top;
+
+  if(top) {
+    pthread_mutex_lock(&A);
+    g = 10;
+    h = 10;
+    pthread_mutex_unlock(&A);
+
+    // If the analysis does unsoundly not account for pthread_exit, these writes are lost
+    pthread_exit(NULL);
```

Comment:
Wait, if no `thread_return` uses `st.apr`, then how are these writes now being accounted for?

Reply:
The key is that the join-local component of the more involved privatizations needs to account for writes to globals after the thread that is excited here is must-joined. For this reason, the `thread_return` side effects the local `lmust,l` information to a special unknown.

https://github.com/goblint/analyzer/blob/829e16a044925b0baf08febf3555572d3271c907/src/analyses/apron/apronPriv.apron.ml#L1064-L1067

Reply:
Right, the `L` component contains the written abstract values, not just `st.apr`.